Fixed mismatch of preview in the pattern of a particular

shiena opened the pull request:
hoogle command because the output is extra message when you search for
a particular query, the preview does not match the candidate.
For this reason, it is skipped extra message.

The following is a specific pattern and its source.

https://github.com/ndmitchell/hoogle/blob/master/src/Hoogle/Query/Suggest.hs

case 1: Search Google

``` output:
Google rocks!
```

case 2: Search ?oogle (ex. hoogle)

``` output:
Can't think of anything more interesting to search for?
```

case 3: Suggestion message of "Did you mean: " (ex. search "IO String")

``` output:
Did you mean: :: IO String
```

case 4: None search result (ex. search "foo-bar-baz")

``` output:
No results found
```

case 5: Conbination of case 2 and case 4 (ex. coogle)

``` output:
Can't think of anything more interesting to search for?
No results found
```

shiena committed:
Fixed mismatch of preview in the pattern of a particular

hoogle command because the output is extra message when you search for
a particular query, the preview does not match the candidate.
For this reason, it is skipped extra message.

The following is a specific pattern and its source.

https://github.com/ndmitchell/hoogle/blob/master/src/Hoogle/Query/Suggest.hs

case 1: Search Google

```output:
Google rocks!
```

case 2: Search ?oogle (ex. hoogle)

```output:
Can't think of anything more interesting to search for?
```

case 3: Suggestion message of "Did you mean: " (ex. search "IO String")

```output:
Did you mean: :: IO String
```

case 4: None search result (ex. search "foo-bar-baz")

```output:
No results found
```

case 5: Conbination of case 2 and case 4 (ex. coogle)

```output:
Can't think of anything more interesting to search for?
No results found
```

eagletmt commented:
Good catch :+1: 

I noticed two points.
1. Isn't it enough to exclude lines that don't contain `-- http://` while gathering candidates?
2. Even after applying your patch, the preview contains unwanted suggestions.

eagletmt commented:
A more appropriate fix to this issue would be adding `--disable-suggestions` flag to hoogle...

shiena committed:
Improved deletion of verbose message

Delete the following message will be added when you add the `verbose` flag
to hoogle command.

Search results are output after the message `= ANSWER =`.
Tags such as `  -- ` and `  -- Exact` and `  -- PrefixCI+Substr` is added
to the search results.

shiena committed:
Fixed a mismatch case of not containing ` -- http://` in the search results

shiena commented:
Thank you for review.

> Isn't it enough to exclude lines that don't contain -- http:// while gathering candidates?

I was fixed mismatch by setting the `action__haddock_index` at the same time as the declaration of: `l:candidate`.
In addition, since the three patterns of `-- http://`, `-- file://`, `--` URL part, and the rest was also supported.

> Even after applying your patch, the preview contains unwanted suggestions.

I was able to fix it so that it does not show candidate in the preview.

eagletmt commented:
Looks good to me.
Thank you for your contribution.
